Source/WebCore/Modules/permissions/Permissions.h

```diff
@@ -36,24 +36,24 @@ class JSObject;
 
 namespace WebCore {
 
-class Navigator;
+class NavigatorBase;
 class PermissionStatus;
+class ScriptExecutionContext;
 
 template<typename IDLType> class DOMPromiseDeferred;
 
 class Permissions : public RefCounted<Permissions> {
     WTF_MAKE_ISO_ALLOCATED(Permissions);
 public:
-    static Ref<Permissions> create(Navigator&);
+    static Ref<Permissions> create(NavigatorBase&);
     ~Permissions();
 
-    Navigator* navigator();
-    void query(JSC::Strong<JSC::JSObject>, DOMPromiseDeferred<IDLInterface<PermissionStatus>>&&);
+    NavigatorBase* navigator();
+    void query(ScriptExecutionContext&, JSC::Strong<JSC::JSObject>, DOMPromiseDeferred<IDLInterface<PermissionStatus>>&&);
 
 private:
-    explicit Permissions(Navigator&);
-
-    WeakPtr<Navigator> m_navigator;
+    explicit Permissions(NavigatorBase&);
+    WeakPtr<NavigatorBase> m_navigator;
```

Comment:
Please add a blank line before the data members, like there used to be.

Reply:
Fixed!